### PR TITLE
WT-8750 Change S3 extension to use CPP conventions

### PR DESCRIPTION
**Pull request by fionaochee**
* Change `.cxx` extension to `.cpp`.
* Change naming from underscore casing to camel case.
* Change class name to `S3Connection`.
* Change store name to `S3StorageSource`.

### CODE REVIEW

File: ext/storage_sources/s3_store/CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 project(s3_store CXX)    
 
 # Create an intermediate library to build the helper library.
-add_library(aws_bucket_util STATIC aws_bucket_conn.cxx)
+add_library(aws_bucket_util STATIC S3Connection.cpp)
```
Comment (sulabhM) [inline ext/storage_sources/s3_store/CMakeLists.txt:4]: Please don't rename the files in the camel case, the new name could be ```s3_connection.cpp``` . Similarly ```s3_storage_source.cpp```

File: ext/storage_sources/s3_store/S3StorageSource.cpp [hunk 2]
```diff
@@ -160,46 +160,46 @@ s3_customize_file_system(WT_STORAGE_SOURCE *storage_source, WT_SESSION *session,
         }
     }
 
-    *file_systemp = &fs->file_system;
+    *fileSystemp = &fs->fileSystem;
```
Comment (sulabhM) [inline ext/storage_sources/s3_store/S3StorageSource.cpp:163]: This looks weird

File: ext/storage_sources/s3_store/S3StorageSource.cpp [hunk 3]
```diff
@@ -160,46 +160,46 @@ s3_customize_file_system(WT_STORAGE_SOURCE *storage_source, WT_SESSION *session,
...
 }
 
 /*
- * s3_fs_terminate --
+ * S3FsTerminate --
  *     Discard any resources on termination of the file system.
  */
 static int
-s3_fs_terminate(WT_FILE_SYSTEM *file_system, WT_SESSION *session)
+S3FsTerminate(WT_FILE_SYSTEM *fileSystem, WT_SESSION *session)
```
Comment (sulabhM) [inline ext/storage_sources/s3_store/S3StorageSource.cpp:172]: This also looks weird.

**Review by sulabhM — APPROVED** [review]

File: ext/storage_sources/s3_store/S3StorageSource.cpp (hunk 2 above)
Comment (tammybailey) [reply ext/storage_sources/s3_store/S3StorageSource.cpp]: We don't need the `p`.

File: ext/storage_sources/s3_store/S3StorageSource.cpp (hunk 3 above)
Comment (tammybailey) [reply ext/storage_sources/s3_store/S3StorageSource.cpp]: `S3FileSystemTerminate`, perhaps?

**Review by tammybailey — APPROVED** [review]

**Pull request merged**